hal/armv8m/nrf/91/console.c

```diff
@@ -0,0 +1,145 @@
+/*
+ * Phoenix-RTOS
+ *
+ * plo - operating system loader
+ *
+ * Console
+ *
+ * Copyright 2022 Phoenix Systems
+ * Authors: Damian Loewnau
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <hal/hal.h>
+#include "nrf91.h"
+#include <board_config.h>
+
+
+/* currently supported baud rates - 9600, 115200 */
+#define BAUD_9600_REGVAL   0x00275000u
+#define BAUD_115200_REGVAL 0x01d60000u
+
+
+static struct {
+	volatile u32 *base;
+	u32 txDma;
+	s32 txDmaSize;
+} halconsole_common;
+
+
+/* clang-format off */
+enum { uarte_startrx = 0, uarte_stoprx, uarte_starttx, uarte_stoptx, uarte_flushrx = 11,
+uarte_events_cts = 64, uarte_events_rxdrdy = 66, uarte_events_endrx = 68, uarte_events_txdrdy = 71,
+uarte_events_endtx, uarte_events_error, uarte_events_rxto = 81, uarte_events_txstarted = 84,
+uarte_inten = 192, uarte_intenset, uarte_intenclr, uarte_errorsrc = 288, uarte_enable = 320,
+uarte_psel_rts = 322, uarte_psel_txd, uarte_psel_cts, uarte_psel_rxd, uarte_baudrate = 329,
+uarte_rxd_ptr = 333, uarte_rxd_maxcnt, uarte_rxd_amount, uarte_txd_ptr = 337, uarte_txd_maxcnt, uarte_txd_amount,
+uarte_config = 347 };
+/* clang-format on */
+
+
+/* Maximum number of characters that will be sent is specified by UART_TX_DMA_SIZE */
+void hal_consolePrint(const char *s)
+{
+	volatile char *tx_dma_buff = (volatile char *)halconsole_common.txDma;
+	int cnt = 0;
+
+	do {
+		tx_dma_buff[cnt] = s[cnt];
+		cnt++;
+	} while (s[cnt - 1] != '\0');
+
+	if (cnt > halconsole_common.txDmaSize) {
+		cnt = halconsole_common.txDmaSize;
+	}
+
+	*(halconsole_common.base + uarte_txd_ptr) = halconsole_common.txDma;
+	*(halconsole_common.base + uarte_txd_maxcnt) = cnt;
+	*(halconsole_common.base + uarte_starttx) = 1u;
+	while (*(halconsole_common.base + uarte_events_txstarted) != 1u) {
+		;
+	}
+
+	*(halconsole_common.base + uarte_events_txstarted) = 0u;
+
+	while (*(halconsole_common.base + uarte_events_endtx) != 1u) {
+		;
+	}
+
+	*(halconsole_common.base + uarte_events_endtx) = 0u;
+}
+
+
+void console_init(void)
+{
+	const struct {
```

Comment:
```suggestion
	static const struct {
```

Reply:
Why do we want to have static storage duration here?